django/db/models/expressions.py

```diff
@@ -651,6 +654,39 @@ def relabeled_clone(self, relabels):
         return self
 
 
+class SliceableF(F):
+    """
+    An object that contains a slice of an F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f_obj, slice_obj):
```

Comment:
I feel `slice_obj` is a bad name to use here given we allow indexing by integer. Perhaps `subscript`?

Maybe then we can just simplify `f_obj` → `obj`?